fedmsg/crypto/x509.py

```diff
@@ -113,7 +113,7 @@ def fail(reason):
     for field in ['signature', 'certificate']:
         if field not in message:
             return fail("No %r field found." % field)
-        if not isinstance(message[field], str):
+        if not isinstance(message[field], basestring):
```

Comment:
``basestring`` is undefined in Python 3. While this code will never actually run in Python 3 since it's the M2Crypto implementation, this makes the API unclear and it should either be ``six.text_type`` (unicode and str in Python 2 and Python 3, respectively) or ``six.binary_type`` (str and bytes in Python 2 and Python 3, respectively).

Since we don't know the encoding at this point, I believe it should be ``six.text_type``.